src/rrt/Tree.hpp

```diff
@@ -218,13 +251,27 @@ class Tree {
     }
 
     /**
-     * Removes all Nodes from the tree so it can be run() again.
+     * Removes nodes from _nodes and _nodemap so it can be run() again.
      */
     void reset(bool eraseRoot = false) {
+        for (int i = 1; i < _kdtree.size(); i++) {
+            _kdtree.removePoint(i);
+        }
         if (eraseRoot) {
+            _kdtree.removePoint(0);
             _nodes.clear();
+            _nodemap.clear();
         } else {
-            if (!_nodes.empty()) {
+            if (_nodes.size() > 1) {
```

Comment:
The logic here is a bit confusing. And the multiple iteration might not be particularly fast. Would it be valid to just store the map entry for the root load. Clear the entire map and then reinsert it? Also just to check, is eraseRoot=False actually used?